ramlfications/loader.py

```diff
@@ -35,7 +39,74 @@ def _yaml_include(self, loader, node):
                 return inputfile.read()
 
         with open(file_name) as inputfile:
-            return yaml.load(inputfile, self._ordered_loader)
+            parsed = yaml.load(inputfile, self._ordered_loader)
+
+            if file_ext == ".json":
+                parsed = self._parse_json_refs(
+                    parsed,
+                    os.path.dirname(file_name),
+                )
+
+            return parsed
+
+    def _lookup_json_ref(self, ref_key, base_path=None, parent_schema=None):
+        """
+        Traverses the json pointer and returns the value of the pointer.
+        """
+        if "#" not in ref_key: raise Exception("Ref values must contain a fragment (#).")
+
+        ref_uri, ref_fragment = ref_key.split("#")
+        # Load the ref and cache it if the ref is not an internal reference
+        if ref_uri not in self.refs.keys():
+            if ref_uri is not '':
+                response = download_url(ref_uri)
+                self.refs[ref_uri] = self._ordered_load(response, yaml.SafeLoader)
+
+
+        # Hack to make the "whole file" be the empty string, which is the
+        # part of the reference fragment before the first slash (or the whole
+        # fragment, if there's no slash). Also, grab the correct schema from the cache.
```

Comment:
Last sentence is a bit of a lie. "...from the cache or use the parent schema" or something?